core/src/main/java/com/novoda/accessibility/ActionMenuItem.java

```diff
@@ -0,0 +1,273 @@
+package com.novoda.accessibility;
+
+import android.content.Intent;
+import android.content.res.Resources;
+import android.graphics.drawable.Drawable;
+import android.support.annotation.Nullable;
+import android.support.annotation.StringRes;
+import android.util.Log;
+import android.view.ActionProvider;
+import android.view.ContextMenu;
+import android.view.MenuItem;
+import android.view.SubMenu;
+import android.view.View;
+
+class ActionMenuItem implements MenuItem {
+
+    private final Resources resources;
+    private final int id;
+
+    private CharSequence title;
+    private boolean enabled;
+    @Nullable
+    private OnMenuItemClickListener menuItemClickListener;
+
+    ActionMenuItem(Resources resources, int id, CharSequence title) {
+        this.resources = resources;
+        this.id = id;
+        this.title = title;
+    }
+
+    @Override
+    public int getItemId() {
+        return id;
+    }
+
+    @Override
+    public int getGroupId() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int getOrder() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ActionMenuItem setTitle(CharSequence title) {
+        this.title = title;
+        return this;
+    }
+
+    @Override
+    public ActionMenuItem setTitle(@StringRes int title) {
+        setTitle(resources.getString(title));
+        return this;
+    }
+
+    @Override
+    public CharSequence getTitle() {
+        return title;
+    }
+
+    @Override
+    public ActionMenuItem setTitleCondensed(@Nullable CharSequence title) {
+        if (title != null) {
+            throw new UnsupportedOperationException();
+        }
+        return this;
+    }
+
+    @Override
+    public CharSequence getTitleCondensed() {
+        return title;
+    }
+
+    @Override
+    public ActionMenuItem setIcon(Drawable icon) {
```

Comment:
We could eventually support this and the next - perhaps the AlertDialog we show could be styled or have a custom item view which displays the icon **and** title.